src/ui/page/dataverse/dataspace/governance/governanceNavigation.tsx

```diff
@@ -28,56 +39,92 @@ export const GovernanceNavigation: FC<GovernanceWithNavigationProps> = ({
 
   const governanceBasePath = `/dataverse/dataspace/${dataspaceId}/governance`
 
+  const convertToOptions = useMemo(
+    () =>
+      (sections: Section[]): Option[] =>
+        sections.map(({ id, title, contains }) => ({
+          id: id,
+          title: title,
+          value: contains && convertToOptions(contains)
+        })),
+    []
+  )
```

Comment:
Is it necessary to have a `Select` component that can accommodate multiple levels of submenu nesting? Could we consider implementing it in a similar manner to the desktop version, with one navigation element for each section or subsection?

While creating a dedicated logic for mobile is understandable due to differing behavior from the desktop version, it appears that we may not require additional nesting levels elsewhere in the app.

Typically, Select menus and Options are associated with forms, although the W3C does not enforce strict guidelines. Perhaps a name that reflects its relation to navigation might be more suitable.

Reply:
As discussed we want to have as generic components as possible. I've reworked the component so it can handle nested values and to allow a flexible selected title. A consequence of this however is the complexity of its implementation as the value of each option represents the menu structure.